script.js

```diff
@@ -0,0 +1,53 @@
+let bookCollection = [];
+
+function displayBook() {
+  bookCollection = JSON.parse(localStorage.getItem('bookCollection')) || { title: '', author: '' };
+  const bookList = document.querySelector('.books-list');
+  bookList.innerHTML = '';
+  for (let i = 0; i < bookCollection.length; i += 1) {
+    const book = bookCollection[i];
+    const bookItem = document.createElement('div');
+    const bookButton = document.createElement('button');
+    bookButton.innerText = 'Remove';
+    bookButton.classList.add('removeBtn');
+    bookButton.setAttribute('id', i);
+    bookItem.innerHTML = `${book.title} </br> ${book.author}`;
+    const horizontalLine = document.createElement('hr');
+    bookList.appendChild(bookItem);
+    bookList.appendChild(bookButton);
+    bookList.appendChild(horizontalLine);
+  }
+  const removeButton = document.querySelectorAll('.removeBtn');
+
+  function removeBook(id) {
+    const newid = Number(id);
+    bookCollection = bookCollection.filter((item, index) => index !== newid);
+    localStorage.setItem('bookCollection', JSON.stringify(bookCollection));
+    displayBook();
+  }
+
+  removeButton.forEach((item) => {
+    item.addEventListener('click', () => {
+      removeBook(item.id);
+    });
+  });
+}
+
+function addBook(title, author) {
+  const newBook = { title, author };
+  bookCollection.push(newBook);
+  localStorage.setItem('bookCollection', JSON.stringify(bookCollection));
```

Comment:
- [ ] I get the following error while trying to add a book:
 
` script.js:38 Uncaught TypeError: bookCollection.push is not a function at addBook (script.js:38:18)
` 

It seems that bookCollection is declared as an empty array. The error message I received suggests that bookCollection is not an array, which could be caused by reassigning bookCollection to a non-array value somewhere in your code.

To fix this issue, you can try adding a console.log statement before the line where bookCollection.push is called to check the value of bookCollection. This can help you identify where bookCollection is being reassigned to a non-array value.

Additionally, you can try adding a check to ensure that bookCollection is an array before calling the push method. Here's an example:

```.js
function addBook(title, author) {
  const newBook = { title, author };
  if (!Array.isArray(bookCollection)) {
    bookCollection = [];
  }
  bookCollection.push(newBook);
  localStorage.setItem('bookCollection', JSON.stringify(bookCollection));
  displayBook();
}
```